### PR TITLE
Add begin/end_transaction methods in FlightSqlServiceClient

### PR DESCRIPTION
**Pull request by lewiszlw**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 

`FlightSqlService` suppots transaction feature, so add some helper methods for client.

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-flight/src/sql/client.rs [hunk 1]
```diff
@@ -399,6 +403,54 @@ impl FlightSqlServiceClient<Channel> {
         ))
     }
 
+    /// Request to begin a transaction.
+    pub async fn begin_transaction(&mut self) -> Result<Bytes, ArrowError> {
```
Comment (alamb) [inline arrow-flight/src/sql/client.rs:407]: I was wondering if there were any parameters that should be passed to this function, but I see the underlying message doesn't have any either so this makes sense to me

**Review by alamb — COMMENTED** [review]
Looks good to me -- thank you @lewiszlw 

Could you please add basic integration tests for this code (that shows the data correctly making it from the client to the server)? 

I think an example of such a test is here.

https://github.com/apache/arrow-rs/blob/eb2d00b3ab225395e664bc578a1b56c42c6ac32d/arrow-flight/tests/client.rs#L46-L70

Though that looks like it is only for `FlightServiceClient` rather than the `FlightSqlServiceClient` 🤔

**Comment by lewiszlw** [comment]
I added a test in `flight_sql_client_cli.rs`. (maybe could move to a new file `flight_sql_client.rs`)

File: arrow-flight/tests/flight_sql_client_cli.rs [hunk 2]
```diff
@@ -318,244 +358,6 @@ impl FlightSqlService for FlightSqlServiceImpl {
         Ok(resp)
     }
 
-    async fn get_flight_info_substrait_plan(
```
Comment (alamb) [inline arrow-flight/tests/flight_sql_client_cli.rs:321]: this all have default impls, so it is ok to remove them, right?

**Comment by alamb** [comment]
>  I added a test in flight_sql_client_cli.rs. (maybe could move to a new file flight_sql_client.rs)

Yeah, it seems like  flight_sql_client_cli.rs is slowly becoming the test for the `FlightSqlServiceClient` -- I agree reaming it and starting to cover all the FlightSQL methods might be good

**Review by alamb — APPROVED** [review]
Thanks @lewiszlw

File: arrow-flight/tests/flight_sql_client_cli.rs (hunk 2 above)
Comment (lewiszlw) [reply arrow-flight/tests/flight_sql_client_cli.rs:321]: Yes.

**Pull request merged**